src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```diff
@@ -209,6 +218,17 @@ internal sealed class SolutionRestoreJob : ISolutionRestoreJob
                     isSolutionAvailable,
                     deferredProjectsData.PackageSpecs,
                     token);
+
+                // To limit risk, we only publish the event when there is a cross-platform PackageReference project
+                // in the solution. Extending this behavior to all solutions is tracked here:
+                // https://github.com/NuGet/Home/issues/4478
+#if VS15
+                if (projects.OfType<CpsPackageReferenceProject>().Any())
+                {
+                    var args = new SolutionRestoredEventArgs(_dependencyGraphProjectCacheHash);
+                    _restoreEventsPublisher.OnSolutionRestoreCompleted(args);
+                }
```

Comment:
If restore fails, there will be no assets file. If no assets file, PMC won't be able to enumerate dependencies (which is required to discover the set of init.ps1 to execute). So, yes.